nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/storage/ListAzureBlobStorage.java

```diff
@@ -176,45 +174,38 @@ protected String getDefaultTimePrecision() {
 
-            final OperationContext operationContext = new OperationContext();
-            AzureStorageUtils.setProxy(operationContext, context);
+            final ListBlobsOptions listBlobsOptions = new ListBlobsOptions()
+                .setPrefix(prefix)
+                .setDetails(new BlobListDetails()
+                    .setRetrieveMetadata(true));
 
-            for (ListBlobItem blob : container.listBlobs(prefix, true, EnumSet.of(BlobListingDetails.METADATA), null, operationContext)) {
-                if (blob instanceof CloudBlob) {
-                    CloudBlob cloudBlob = (CloudBlob) blob;
-                    BlobProperties properties = cloudBlob.getProperties();
-                    StorageUri uri = cloudBlob.getSnapshotQualifiedStorageUri();
+            blobContainerClient.listBlobs().forEach(blob -> {
+                if (blob instanceof BlobItem) {
+                    BlobItem blobItem = (BlobItem) blob;
+                    BlobItemProperties properties = blobItem.getProperties();
+                    BlobClient blobClient = blobContainerClient.getBlobClient(blobItem.getName());
+                    String uri = blobClient.getBlobUrl();
 
                     Builder builder = new BlobInfo.Builder()
-                                              .primaryUri(uri.getPrimaryUri().toString())
-                                              .blobName(cloudBlob.getName())
-                                              .containerName(containerName)
+                                              .primaryUri(uri)
+                                              .blobName(blobItem.getName())
+                                              .blobType(properties.getBlobType().toString())
+                                              .containerName(blobClient.getContainerName())
                                               .contentType(properties.getContentType())
                                               .contentLanguage(properties.getContentLanguage())
-                                              .etag(properties.getEtag())
-                                              .lastModifiedTime(properties.getLastModified().getTime())
-                                              .length(properties.getLength());
-
-                    if (uri.getSecondaryUri() != null) {
-                        builder.secondaryUri(uri.getSecondaryUri().toString());
-                    }
-
-                    if (blob instanceof CloudBlockBlob) {
-                        builder.blobType(AzureStorageUtils.BLOCK);
-                    } else {
-                        builder.blobType(AzureStorageUtils.PAGE);
-                    }
+                                              .etag(properties.getETag())
+                                              .lastModifiedTime(properties.getLastModified().toEpochSecond())
+                                              .length(properties.getContentLength());
+
```

Comment:
Secondary URI is no longer receivable from v12 onwards. Would you suggest we keep it in here as a backwards compatible output (but it will be empty anyways...) or just remove it from BlobInfo class?